src/components/layout/BottomModal.tsx

```diff
@@ -133,61 +133,77 @@ const BottomModal = ({ Content, isOpen, onClose, scrollableContent, customMinHei
     }
   }
 
-  const handleClose = () => {
+  const handleClose = useCallback(() => {
     'worklet'
 
     navHeight.value = withSpring(0, springConfig)
     modalHeight.value = withSpring(0, springConfig, (finished) => finished && runOnJS(onClose)())
     position.value = 'closing'
-  }
+  }, [modalHeight, navHeight, onClose, position])
 
-  const handleMaximize = () => {
+  const handleMaximize = useCallback(() => {
     'worklet'
 
     navHeight.value = withSpring(NAV_HEIGHT + 10, springConfig)
     modalHeight.value = withSpring(-maxHeight, springConfig)
     position.value = 'maximised'
-  }
+  }, [maxHeight, modalHeight, navHeight, position])
 
-  const handleMinimize = () => {
+  const handleMinimize = useCallback(() => {
     'worklet'
 
     navHeight.value = withSpring(0, springConfig)
     modalHeight.value = withSpring(-minHeight.value, springConfig)
     position.value = 'minimised'
-  }
-
-  const panGesture = Gesture.Pan()
-    .onStart((e) => {
-      offsetY.value = modalHeight.value
-    })
-    .onChange((e) => {
-      if (position.value !== 'closing') {
-        modalHeight.value = offsetY.value + e.translationY
-      }
-    })
-    .onEnd(() => {
-      const shouldMinimise = position.value === 'maximised' && -modalHeight.value < dimensions.height - DRAG_BUFFER
-
-      const shouldMaximise =
-        canMaximize.value && position.value === 'minimised' && -modalHeight.value > minHeight.value + DRAG_BUFFER
-
-      const shouldClose =
-        ['minimised', 'closing'].includes(position.value) && -modalHeight.value < minHeight.value - DRAG_BUFFER
-
-      if (shouldMaximise) {
-        handleMaximize()
-      } else if (shouldMinimise) {
-        scrollableContent ? handleClose() : handleMinimize()
-      } else if (shouldClose) {
-        handleClose()
-      } else {
-        modalHeight.value =
-          position.value === 'maximised'
-            ? withSpring(-maxHeight, springConfig)
-            : withSpring(-minHeight.value, springConfig)
-      }
-    })
+  }, [minHeight.value, modalHeight, navHeight, position])
+
+  const panGesture = useMemo(
```

Comment:
Following the tip from here: https://docs.swmansion.com/react-native-gesture-handler/docs/api/gestures/gesture/

Reply:
Ah, that's great!